libraries/AP_Networking/AP_Networking_tests.cpp

```diff
@@ -113,10 +113,10 @@ void AP_Networking::test_TCP_discard(void)
         return;
     }
     // connect to the discard service, which is port 9
-    if (!sock->connect(dest, 9)) {
-        GCS_SEND_TEXT(MAV_SEVERITY_ERROR, "TCP_discard: connect failed");
-        return;
+    while (!sock->connect(dest, 9)) {
+        hal.scheduler->delay(10);
```

Comment:
timeout?

Reply:
I deliberately have this wait forever, quite useful when running tests